backend/src/bundles/friends/friend.repository.ts

```diff
@@ -0,0 +1,191 @@
+
+        const nonFriendUsers = allUsers.filter(
+            (user) => !followingIds.includes(user.id) && user.id !== userId,
+        );
+
+        const paginatedUsers = nonFriendUsers
+            .slice(Number(offset), Number(offset) + Number(limit))
+            .map((user) => {
+                const { userDetails, id, email } = user;
+                return {
+                    id,
+                    userId: userDetails.userId,
+                    email,
+                    avatarUrl: userDetails.avatarUrl,
+                    username: userDetails.username,
+                    fullName: userDetails.fullName,
+                    dateOfBirth: userDetails.dateOfBirth,
+                    weight: userDetails.weight,
+                    height: userDetails.height,
+                    location: userDetails.location,
+                    gender: userDetails.gender,
+                };
+            });
+
+        return paginatedUsers as FriendResponseDto[];
+    }
+
+    public async getFollowings(
+        userId: number,
+        offset: string,
+        limit: string,
+    ): Promise<FriendResponseDto[] | null> {
+        const followings = await FriendModel.query()
+            .select(`${DatabaseTableName.USERS}.email`, 'followingId')
+            .where(`${DatabaseTableName.FRIENDS}.userId`, userId)
+            .withGraphFetched('userDetails')
+            .join(
+                DatabaseTableName.USERS,
+                `${DatabaseTableName.USERS}.id`,
+                `${DatabaseTableName.FRIENDS}.followingId`,
+            )
+            .orderBy(`${DatabaseTableName.FRIENDS}.id`)
+            .offset(Number(offset))
+            .limit(Number(limit));
+
+        return followings.map(({ id, userDetails, email }) => ({
+            id,
+            userId: userDetails.userId,
+            email,
+            avatarUrl: userDetails.avatarUrl,
+            username: userDetails.username,
+            fullName: userDetails.fullName,
+            dateOfBirth: userDetails.dateOfBirth,
+            weight: userDetails.weight,
+            height: userDetails.height,
+            location: userDetails.location,
+            gender: userDetails.gender,
+        })) as FriendResponseDto[];
+    }
+
+    public async addFollowing(
+        id: number,
+        followingId: number,
+        offset: string,
+    ): Promise<FriendResponseDto | null> {
+        const trx = await this.userModel.startTransaction();
+
+        const user = await this.userModel.query(trx).findById(id);
+        if (!user) {
+            throw new Error('User not found');
+        }
+
+        await user
+            .$relatedQuery('friends', trx)
+            .insert({ followingId })
+            .returning('*')
+            .first();
+
+        const result = await this.findAllPotentialFollowings(id, offset, '1');
+
+        await trx.commit();
+        return result as unknown as FriendResponseDto;
```

Comment:
why as unknown?

Reply:
seems like return type is array or null. why are you trying to cast it to single dto instance? I don't get it

Reply:
fixed